datasets/pl/finanse_sanctions/crawler.py

```diff
@@ -16,88 +16,107 @@
     entity.add("birthCountry", birth_country, lang="pol")
-    h.apply_date(entity, "birthDate", row.pop("Data urodzenia"))
-    entity.add("address", row.pop("location full") or None)
-    entity.add("country", row.pop("location country") or None)
+    h.apply_date(entity, "birthDate", row.pop("data_urodzenia"))
 
-    entity.add("nationality", row.pop("narodowość"))
-    entity.add("topics", "sanction")
+    other_information = row.pop("inne_informacje")
+    res = context.lookup("other_information", other_information)
+    if not res:
+        context.log.warning("No extra information lookup found", row=row)
+    else:
+        entity.add(
+            "nationality",
+            res.properties.get("nationality"),
+            lang="pol",
+            original_value=other_information,
+        )
+        entity.add(
+            "country",
+            res.properties.get("country"),
+            lang="pol",
+            original_value=other_information,
+        )
+        entity.add(
+            "address",
+            res.properties.get("address"),
+            lang="pol",
+            original_value=other_information,
+        )
 
+    entity.add("topics", "sanction")
     sanction = h.make_sanction(context, entity)
-    h.apply_date(sanction, "listingDate", row.pop("Data umieszczenia na liście"))
+    h.apply_date(sanction, "listingDate", row.pop("data_umieszczenia_na_liscie"))
     sanction.add(
-        "reason", collapse_spaces(row.pop("Uzasadnienie wpisu na listę")), lang="pol"
+        "reason",
+        collapse_spaces(row.pop("uzasadnienie_wpisu_na_liste_url")),
+        lang="pol",
     )
     sanction.add("program", POLAND_PROGRAM, "pol")
 
     context.emit(entity, target=True)
     context.emit(sanction)
 
-    context.audit_data(row)
+    context.audit_data(
+        row,
+        ignore=[
+            "lp",  # ID, we don't trust this to be stable
+            "uzasadnienie_wpisu_na_liste",  # Reason link text, we only care about the url
+        ],
+    )
 
 
-def check_updates(context: Context):
+def crawl(context: Context):
     doc = context.fetch_html(context.dataset.url)
     doc.make_links_absolute(context.dataset.url)
-    materials = doc.findall(".//a[@class='file-download']")
 
-    # Process the materials
-    if len(materials) == 0:
-        context.log.warning("No materials downloads found")
-    else:
-        for material in materials:
-            url = material.get("href")
-            if url in KNOWN_HASHES:
-                h.assert_url_hash(context, url, KNOWN_HASHES[url])
-            else:
-                context.log.warning(
-                    "Unknown materials download URL. Check if we want to scrape it.",
-                    url=url,
-                )
+    xlsx_link_element = doc.xpath(
+        ".//a[@class='file-download' and contains(text(), 'Lista osób i podmiotów, wobec których stosuje się szczególne środki ograniczające na podstawie art. 118 ust. 1 pkt 2 (wersja xlsx)')]"
```

Comment:
Maybe this is sufficient?

```suggestion
        ".//a[@class='file-download' and contains(text(), '118 ust. 1 pkt 2 (wersja xlsx)')]"
```